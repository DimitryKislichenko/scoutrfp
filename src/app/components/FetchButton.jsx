import React from 'react';
import { Button } from 'semantic-ui-react';

const FetchButton = ({ onClick }) => (
    <Button onClick={onClick} size="big" primary fluid>
        Fetch Rates
    </Button>
);

export default FetchButton;
