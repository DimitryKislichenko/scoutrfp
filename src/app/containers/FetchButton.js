import { connect } from 'react-redux';

import FetchButton from '../components/FetchButton';

// An action to trigger rates fetching
const fetchRates = () => ({
    type: 'FETCH_RATES',
});

export default connect(
    ({ rates }) => ({ loading: rates.loading }),
    { onClick: fetchRates },
)(FetchButton);
