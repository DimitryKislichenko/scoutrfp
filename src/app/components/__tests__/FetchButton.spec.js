import React from 'react';
import Enzyme, { render, shallow } from 'enzyme';
import enzymeAdapterReact16 from 'enzyme-adapter-react-16';

import FetchButton from '../FetchButton';

Enzyme.configure({ adapter: new enzymeAdapterReact16() });

describe('FetchButton', () => {
    it('should render without crashing', () => {
        shallow(<FetchButton />);
    });

    it('should be disable while loading data', () => {
        const component = render(<FetchButton loading />);
        expect(component).toMatchSnapshot();
    });

    it('should trigger onClick event when button is clicked', () => {
        const onClick = jest.fn();
        const component = shallow(<FetchButton onClick={onClick} />);
        component.simulate('click');

        expect(onClick).toBeCalled();
    });
});
