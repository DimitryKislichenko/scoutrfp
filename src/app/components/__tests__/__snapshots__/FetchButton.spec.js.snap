// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FetchButton should be disable while loading data 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "ui big fluid primary disabled button",
      "disabled": "",
      "role": "button",
      "tabindex": "-1",
    },
    "children": Array [
      Object {
        "data": "Fetch Rates",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "text",
      },
    ],
    "name": "button",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
      "disabled": undefined,
      "role": undefined,
      "tabindex": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
      "disabled": undefined,
      "role": undefined,
      "tabindex": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
